tests/integration-tests/tests/networking/test_placement_group.py

```diff
@@ -0,0 +1,196 @@
+            ):
+                logging.info("Checking placement group for head node")
+                # if the placement type is cluster, make sure the head node also inside the placement group
+                head_node_instance_id = remote_command_executor.run_remote_command(
+                    "curl http://169.254.169.254/latest/meta-data/instance-id"
+                ).stdout
+                head_node_placement_group = boto3.client("ec2", region_name=region).describe_instances(
+                    InstanceIds=[head_node_instance_id]
+                )["Reservations"][0]["Instances"][0]["Placement"]["GroupName"]
+                assert_that(head_node_placement_group).is_equal_to(placement_group)
+        except (NoOptionError, NoSectionError):
+            pass
+
+
+def assert_placement_group_in_queue(cluster, scheduler, region, placement_group, remote_command_executor):
+    logging.info("Checking placement group in queue section")
+    actual_placement_group = get_placement_group(scheduler, region, cluster)
+    assert_that(actual_placement_group).is_equal_to(placement_group)
+
+    if (
+        cluster.config.get("cluster default", "placement")
+        and cluster.config.get("cluster default", "placement") == "cluster"
+    ):
+        logging.info("Checking placement group for head node")
+        # if the placement type is cluster, make sure the head node also inside the placement group
+        head_node_instance_id = remote_command_executor.run_remote_command(
+            "curl http://169.254.169.254/latest/meta-data/instance-id"
+        ).stdout
+        head_node_placement_group = boto3.client("ec2", region_name=region).describe_instances(
+            InstanceIds=[head_node_instance_id]
+        )["Reservations"][0]["Instances"][0]["Placement"]["GroupName"]
+        assert_that(head_node_placement_group).is_equal_to(placement_group)
+
+
+def get_placement_group(scheduler, region, cluster, queue_name):
+    """
+    For slurm, the information of placement group can be retrieved from launch template, for traditional scheduler,
+    the information of placement group can be retrieved from auto-scaling group.
+    """
+    if scheduler == "slurm":
+        ec2_client = boto3.client("ec2", region_name=region)
+        launch_template_name = get_launch_template_name(cluster, queue_name)
+        launch_template_data = ec2_client.describe_launch_template_versions(
+            LaunchTemplateName=launch_template_name, Versions=["$Latest"]
+        )["LaunchTemplateVersions"][0]["LaunchTemplateData"]
+        placement_group_name = launch_template_data["Placement"]["GroupName"]
+        return placement_group_name
+    else:
+        asg_conn = boto3.client("autoscaling", region_name=region)
+        asg_name = get_auto_scalling_group_name(cluster, asg_conn)
+        placement_group_name = asg_conn.describe_auto_scaling_groups(AutoScalingGroupNames=[asg_name])[
+            "AutoScalingGroups"
+        ][0]["PlacementGroup"]
```

Comment:
This looks very similar to `_get_asg` from `scaling_common.py`. Can we make `_get_asg` more generic so you can also use it here, and avoid writing the same code twice?